Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -387,19 +435,220 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
+  refine ⟨As, ?_, ?_, ?_, ?_, ?_⟩
+  · exact MeasurableSet.disjointed (fun n ↦ measurableSet_ball)
+  · exact fun n ↦ Bornology.IsBounded.subset isBounded_ball <| disjointed_subset Bs n
+  · intro n
+    apply (diam_mono (disjointed_subset Bs n) isBounded_ball).trans
+    convert diam_ball half_ε_pos.le
+    ring
+  · have aux : ⋃ n, Bs n = univ := by
+      convert DenseRange.iUnion_uniformity_ball xs_dense <| Metric.dist_mem_uniformity half_ε_pos
+      exact (ball_eq_ball' _ _).symm
+    simpa only [← aux] using iUnion_disjointed
+  · exact disjoint_disjointed Bs
+
+variable {Ω}
+
+lemma ProbabilityMeasure.toMeasure_add_pos_gt_mem_nhds (P : ProbabilityMeasure Ω)
+    {G : Set Ω} (G_open : IsOpen G) {ε : ℝ≥0∞} (ε_pos : 0 < ε) (ε_ne_top : ε ≠ ∞) :
+    ({Q | P.toMeasure G < Q.toMeasure G + ε} ∈ 𝓝 P) := by
+  by_cases easy : P.toMeasure G < ε
+  · exact eventually_of_forall (fun _ ↦ lt_of_lt_of_le easy le_add_self)
+  simp only [not_lt] at easy
+  have aux : P.toMeasure G - ε < liminf (fun Q ↦ Q.toMeasure G) (𝓝 P) := by
+    apply lt_of_lt_of_le (ENNReal.sub_lt_self (measure_lt_top _ _).ne _ _)
+        <| ProbabilityMeasure.le_liminf_measure_open_of_tendsto tendsto_id G_open
+    · exact (lt_of_lt_of_le ε_pos easy).ne.symm
+    · exact ε_pos.ne.symm
+  filter_upwards [gt_mem_sets_of_limsInf_gt (α := ℝ≥0∞) isBounded_ge_of_bot
+      (show P.toMeasure G - ε < limsInf ((𝓝 P).map (fun Q ↦ Q.toMeasure G)) from aux)] with Q hQ
+  simp only [preimage_setOf_eq, mem_setOf_eq] at hQ
+  convert ENNReal.add_lt_add_right ε_ne_top hQ
+  exact (tsub_add_cancel_of_le easy).symm
+
+lemma ProbabilityMeasure.continuous_toLevyProkhorov :
+    Continuous (ProbabilityMeasure.toLevyProkhorov (Ω := Ω)) := by
+
+  -- We will check continuity of `id : ProbabilityMeasure Ω → LevyProkhorov (ProbabilityMeasure Ω)`
+  -- at each point `P : ProbabilityMeasure Ω`.
+  rw [continuous_iff_continuousAt]
+  intro P
+
+  -- To check continuity, fix `ε > 0`. To leave some wiggle room, be ready to use `ε/3 > 0` instead.
+  rw [continuousAt_iff']
+  intro ε ε_pos
+  have third_ε_pos : 0 < ε / 3 := by linarith
+  have third_ε_pos' : 0 < ENNReal.ofReal (ε / 3) := ofReal_pos.mpr third_ε_pos
+
+  -- First use separability to choose a countable partition of `Ω` into measurable
+  -- subsets `Es n ⊆ Ω` of small diamater, `diam (Es n) < ε/3`.
+  obtain ⟨Es, Es_mble, Es_bdd, Es_diam, Es_cover, Es_disjoint⟩ :=
+    SeparableSpace.exists_measurable_partition_diam_le Ω third_ε_pos
+
+  -- Instead of the whole space `Ω = ⋃ n ∈ ℕ, Es n`, focus on a large but finite
+  -- union `⋃ n < N, Es n`, chosen in such a way that the complement has small `P`-mass,
+  -- `P (⋃ n < N, Es n)ᶜ < ε/3`.
+  have exhaust :=
+    @tendsto_measure_biUnion_Ici_zero_of_pairwise_disjoint Ω _ P.toMeasure _ Es Es_mble Es_disjoint
+  simp only [tendsto_atTop_nhds, Function.comp_apply] at exhaust
+  obtain ⟨N, hN⟩ := exhaust (Iio (ENNReal.ofReal (ε / 3))) third_ε_pos' isOpen_Iio
+  specialize hN N le_rfl
```

Comment:
With `obtain`, you can make your proof more readable by making explicit the properties of `N` you will use, and also making it clear that this is an atomic proof step:
```suggestion
  obtain ⟨N, hN⟩ : ∃ N, (P : Measure Ω) (⋃ i ≥ N, Es i) < ENNReal.ofReal (ε / 3) := by
    have exhaust :=
      @tendsto_measure_biUnion_Ici_zero_of_pairwise_disjoint Ω _ P.toMeasure _ Es Es_mble Es_disjoint
    simp only [tendsto_atTop_nhds, Function.comp_apply] at exhaust
    obtain ⟨N, hN⟩ := exhaust (Iio (ENNReal.ofReal (ε / 3))) third_ε_pos' isOpen_Iio
    exact ⟨N, hN N le_rfl⟩ 
```